{{ cookiecutter.project_slug }}/.pre-commit-config.yaml

```diff
@@ -19,29 +19,22 @@ repos:
       - id: pretty-format-json
         args: ['--autofix', '--indent 2', ]
   - repo: https://github.com/codespell-project/codespell
-    rev: v2.2.4
+    rev: v2.3.0
     hooks:
       - id: codespell
+        args: [ "--toml=api/pyproject.toml" ]
+        additional_dependencies:
+          - tomli
 
   # API
-  - repo: https://github.com/charliermarsh/ruff-pre-commit
-    rev: v0.0.265
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.6.9
     hooks:
       - id: ruff
-        args: ['--fix', '--exit-non-zero-on-fix']
+        args: [ '--fix', '--exit-non-zero-on-fix' ]
+      - id: ruff-format
   - repo: https://github.com/Lucas-C/pre-commit-hooks-safety
-    rev: v1.3.1
+    rev: v1.3.3
     hooks:
       - id: python-safety-dependencies-check
-        entry: safety
-        args: [check]
-        files: ./api/requirements-build.txt
-# TODO: Run this hook from docker container.
-#  - repo: local
-#    hooks:
-#      - id: migrations-check
-#        language: system
-#        name: Check for uncreated migrations.
-#        entry: sh -c "make api-check-migrations"
-#        files: models/*.*\.py$
-#        stages: [commit]
```

Comment:
We will run this on CI, so I think it's OK to remove it from the `pre-commit`

Reply:
I don't think that pre-commit hook & CI are for the same purpose. In my point of view pre-commit hook is a helper for developer to fix and notify some stupid issues like formatting, unused imports & not created migrations.

Pre-commit hook is not required, while CI is. So, pre-commit hook helps to avoid unnecessary CI runs in case of some issues. It's better to get a notification about these issues when you make a commit instead of waiting for a CI run.

I would like to keep this hook.

Reply:
I don't know either why it's commented out.

@SergeyKubrak do you remember the reason why it was commented [here](https://github.com/django-stars/backend-skeleton/commit/0d8cd5cd86be033dfb29e55de24ac5f06d1d8bf2)?

@bulya Please try to uncomment and test it, if you have a capacity. If it won't work, let's try to fix it in another task. Thank you.

Reply:
@bulya It's not a strong requirement. You can remove checking migrations from pre-commit hooks, if it produces additional complexity in CI workflow. It's more important to have this check in CI rather than in pre-commit hooks.

Reply:
I am leaving this for visibility.
We have nevertheless decided to remove this from the pre-commit hooks, as we run all pre-commit hooks in the `linter` CI job, and we don't want either to run it there via the docker-compose or explicitly exclude it from the hooks list.